### ISSUE TITLE
Can you provide code equivalent with yolov4 as the base detector?

### ISSUE THREAD

**Issue by jieruyao49** [YoruCathy/USDSeg-FCOS#3]

**Comment by ElectronicElephant** [comment]
Hi, 

Our Yolov4 implementation is based on a internal project, which will stay close-sourced for another few months.

Also, I have noticed that mmlab is going to officially support Yolov4 (https://github.com/open-mmlab/mmdetection/pull/3781) . If it's merged, I'll add USDSeg based on it as soon as possible.

**Comment by jieruyao49** [comment]
> Hi,
> 
> Our Yolov4 implementation is based on a internal project, which will stay close-sourced for another few months.
> 
> Also, I have noticed that mmlab is going to officially support Yolov4 ([open-mmlab/mmdetection#3781](https://github.com/open-mmlab/mmdetection/pull/3781)) . If it's merged, I'll add USDSeg based on it as soon as possible.

I'm trying to add a mask branch to yolo4 recently, but it's a bit difficult for me. Do you have any good suggestions?

**Comment by ElectronicElephant** [comment]
> > Hi,
> > Our Yolov4 implementation is based on a internal project, which will stay close-sourced for another few months.
> > Also, I have noticed that mmlab is going to officially support Yolov4 ([open-mmlab/mmdetection#3781](https://github.com/open-mmlab/mmdetection/pull/3781)) . If it's merged, I'll add USDSeg based on it as soon as possible.
> 
> I'm trying to add a mask branch to yolo4 recently, but it's a bit difficult for me. Do you have any good suggestions?

Hi, 

The first thing to modify is the dataloader, since most YOLOv4 implementation only provides gt for detection (i.e., bboxes). You need to load gt masks, and generate coefs accordingly.

Then, we can modify the network structures. This part is easy. You can just 'double' the bbox branch in head, but change the `out_features` parameter of the last conv layer to `num_bases * 3` (for bbox branch, it's `(num_classes+4+1)*3` if I'm correct).

Finally, we can modify the loss function part. The pred and gt match of masks stay the same with the ones of bboxes. However, it's still tricky, I have to admit that.

Bests,
